azero/contracts/most/lib.rs

```diff
@@ -377,10 +381,16 @@ pub mod most {
                 )?;
 
                 // bootstrap account with pocket money
-                // NOTE: we don't revert on a failure!
-                _ = self
-                    .env()
-                    .transfer(dest_receiver_address.into(), data.pocket_money);
+                if data.pocket_money_balance >= data.pocket_money {
+                    // don't revert if the transfer fails
+                    _ = self
+                        .env()
+                        .transfer(dest_receiver_address.into(), data.pocket_money);
+                    data.pocket_money_balance = data
+                        .pocket_money_balance
+                        .checked_sub(data.pocket_money)
+                        .ok_or(MostError::Arithmetic)?;
```

Comment:
Question: I think this is inconsistent, no? We don't revert if the transfer fails, but we revert if this subtraction fails? I guess the point is that if this fails, then we have a bug in the math somewhere? Overall, now that I'm seeing this solution, I'm somewhat concerned - it introduces these two pieces of state that must be kept consistent with each other, but there is nothing that guarantees it - for example, in case of some refactoring. Perhaps we should have a struct encapsulating the two fields that makes sure we always update them correctly if you want to keep track of both values? I'm genuinely unsure of what's the best course of action here.

Reply:
Yes, failure here signals a bug in the contract, and for the transfer to fail there would also need to be a bug, so I think I get your point about inconsistency. Do you think both these should revert on failure?

Can you name these two pieces of state? Do you mean the part that keeps track of the rewards and the part that keeps track of pocket money balance? They are both updated along with value transfer/receive which I think is strong enough (unless the current code has some bug that I don't see). I am not sure if I get your point here